Code/Framework/AzCore/AzCore/DOM/DomValue.cpp

```diff
@@ -0,0 +1,1241 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomValue.h>
+#include <AzCore/DOM/DomValueWriter.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+namespace AZ::Dom
+{
+    template<class T>
+    AZStd::shared_ptr<T>& CheckCopyOnWrite(AZStd::shared_ptr<T>& refCountedPointer)
+    {
+        if (refCountedPointer.use_count() > 1)
+        {
+            AZStd::shared_ptr<T> newPointer = AZStd::allocate_shared<T>(AZStdAlloc<ValueAllocator>());
+            *newPointer = *refCountedPointer;
+            refCountedPointer = AZStd::move(newPointer);
+        }
+        return refCountedPointer;
+    }
+
+    Node::Node(AZ::Name name)
+        : m_name(name)
+    {
+    }
+
+    AZ::Name Node::GetName() const
+    {
+        return m_name;
+    }
+
+    void Node::SetName(AZ::Name name)
+    {
+        m_name = name;
```

Comment:
This should use `AZStd::move(name)` as well.
```suggestion
        m_name = AZStd::move(name);
```